### PR TITLE
clippy fixes

### PR DESCRIPTION
**Pull request by jfinkels**
Found using the following command (which appears in `.pre-commit.yaml`):
```
cargo +nightly clippy --workspace --all-targets --all-features --
```

### CODE REVIEW

File: src/uu/echo/src/echo.rs [hunk 1]
```diff
@@ -59,6 +59,10 @@ fn parse_code(
     std::char::from_u32(ret)
 }
 
+// Clippy complatins that `.peek()` is never called on
+// `iter`. However, it is called inside the `parse_code()` function
+// that borrows `iter`.
+#[allow(clippy::unused_peekable)]
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:65]: we are getting this here:
```
ERROR: `cargo clippy`: unknown lint: `clippy::unused_peekable` (file:'src/uu/echo/src/echo.rs', line:65)
```

I guess we don't use the latest version in the CI

File: src/uu/echo/src/echo.rs [hunk 2]
```diff
@@ -59,6 +59,10 @@ fn parse_code(
     std::char::from_u32(ret)
 }
 
+// Clippy complatins that `.peek()` is never called on
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:62]: complains ? :)

**Comment by tertsdiepraam** [comment]
Maybe we should remove the `+nightly` from the `.pre-commit.yaml`? I think it's best to wait for the lints to stabalize and have the false positives removed.

File: src/uu/echo/src/echo.rs (hunk 1 above)
Comment (jfinkels) [reply src/uu/echo/src/echo.rs:65]: I'll just remove this change for now.

**Comment by jfinkels** [comment]
> Maybe we should remove the `+nightly` from the `.pre-commit.yaml`? I think it's best to wait for the lints to stabalize and have the false positives removed.

That's fine with me, I don't have any strong opinion about this.

I have updated this pull request to remove the unstable `#[allow()]` and corrected the spelling.

**Pull request merged**